pkg/controller/controller_utils.go

```diff
@@ -133,10 +133,10 @@ type ControllerExpectations struct {
 
 // GetExpectations returns the ControlleeExpectations of the given controller.
 func (r *ControllerExpectations) GetExpectations(controllerKey string) (*ControlleeExpectations, bool, error) {
-	if exp, exists, err := r.GetByKey(controllerKey); err == nil && exists {
+	if exp, exists, _ := r.GetByKey(controllerKey); exists {
```

Comment:
> So,nothing should be modified?

Right.  I'd leave the existing conditions alone.

I like the additional logging and the typo fix though.